config-gen/src/main/java/org/wildfly/galleon/plugin/config/generator/ForkedConfigGenerator.java

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wildfly.galleon.plugin.config.generator;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Arrays;
+
+import org.jboss.dmr.ModelNode;
+import org.wildfly.galleon.plugin.WfConstants;
+import org.wildfly.galleon.plugin.server.ForkCallback;
+import org.wildfly.galleon.plugin.server.ConfigGeneratorException;
+
+public class ForkedConfigGenerator extends BaseConfigGenerator implements ForkCallback {
+
+   private final ForkCallback callback;
+
+   public ForkedConfigGenerator() {
+      this.callback = null;
+   }
+
+   @Override
+   public void forkedForEmbedded(String... args) throws ConfigGeneratorException {
+      if(args.length != 2) {
+         throw new IllegalArgumentException("Expected one argument but received " + Arrays.asList(args));
+      }
+      this.jbossHome = args[0];
+      final Path script = Paths.get(args[1]);
+      if(!Files.exists(script)) {
+         throw new ConfigGeneratorException("Errors.pathDoesNotExist(script)");
+      }
+      try {
+         executeScript(script);
+      } catch(Exception e) {
```

Comment:
Only IOException should be caught, a ConfigGeneratorException could have been raised.